packages/translator-default/src/analyze.js

```diff
@@ -0,0 +1,54 @@
+import { types as t } from "@marko/babel-types";
+import { isNativeTag, getTagDef } from "@marko/babel-utils";
+
+export const staticNodes = new WeakSet();
+
+export const visitor = {
+  MarkoText(path) {
+    staticNodes.add(path.node);
+  },
+  MarkoPlaceholder(path) {
+    const { confident } = path.get("value").evaluate();
+    if (confident && path.node.escape) staticNodes.add(path.node);
```

Comment:
Probably best to guard the work of `path.evaluate` with a `!path.node.escape` check since that method is semi expensive.